indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractBatchIndexTask.java

```diff
@@ -524,19 +523,16 @@ public static Function<Set<DataSegment>, Set<DataSegment>> compactionStateAnnota
                                       ? null
                                       : new DimensionsSpec(ingestionSpec.getDataSchema().getDimensionsSpec().getDimensions());
       // We only need to store filter since that is the only field auto compaction support
-      Map<String, Object> transformSpec = ingestionSpec.getDataSchema().getTransformSpec() == null || TransformSpec.NONE.equals(ingestionSpec.getDataSchema().getTransformSpec())
+      TransformSpec transformSpec = ingestionSpec.getDataSchema().getTransformSpec() == null || TransformSpec.NONE.equals(ingestionSpec.getDataSchema().getTransformSpec())
```

Comment:
Nit: Pull `ingestionSpec.getDataSchema().getTransformSpec()` into a variable. It is used thrice and seems too verbose.